command.go

```diff
@@ -72,12 +73,21 @@ func newCommandReader(stream string, command *Command) *commandReader {
 	}
 }
 
-func fetchOutput(command *Command) {
+func fetchOutput(ctx context.Context, command *Command) {
 	for {
+		ctxDone := ctx.Done()
 		select {
 		case <-command.cancel:
+			_, _ = command.slurpAllOutput()
+			err := errors.New("canceled")
+			command.Stderr.write.CloseWithError(err)
+			command.Stdout.write.CloseWithError(err)
 			close(command.done)
 			return
+		case <-ctxDone:
```

Comment:
you can write:
```go
case <-ctx.Done():
...
```

Reply:
This is an optimization. We don't exit the loop here, so this would fire every iteration through the loop. With the code as it is, it only fires a single time, then we set `ctxDone` to `nil` and `Close()` the command.

Reply:
Ok, now it really does what I intended it to do after I pulled the definition of `ctxDone` out of the loop.